### PR TITLE
Bigquery table connection

### PR DESCRIPTION
**Pull request by gutschenritter**
Add connection_id field to bigquery table external_data_configuration in order to support Big Lake tables. Fixes #hashicorp/terraform-provider-google/issues/11741.

If this PR is for Terraform, I acknowledge that I have:

- [ X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/third_party/terraform/tests) (for handwritten resources or update tests).
- [X ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
bigquery: Added connection_id to external_data_configurationfor `google_bigquery_table`
```

### CODE REVIEW

**Comment by gutschenritter** [comment]
`make lint` is failing but I think it's because I have a different version installed. 

I'm running into issues with the acceptance tests as well. I'm getting these errors (including for tests I didn't write - just for all tests that create a storage bucket):
```
        Error: googleapi: Error 412: Request violates constraint 'constraints/storage.uniformBucketLevelAccess', conditionNotMet
        
          with google_storage_bucket.test,
          on terraform_plugin_test.tf line 5, in resource "google_storage_bucket" "test":
           5: resource "google_storage_bucket" "test" {
```

I'm running my tests using an internal google.com project, and from what I've seen from internal searches it looks like it may be due to certain internal Overground policies? I'm hoping presubmit runs pass here instead.

**Comment by gutschenritter** [comment]
Looks like there were some actual failures in the tests, and I found another not-overground project I can test from. Will address on Mon.

**Comment by gutschenritter** [comment]
Looks like the lint checks are failing in git but not sure how to debug this - the "details" link takes me to a cloudbuild page & I'm not sure how to parse the info to find out what's wrong with the format. (again, I think I have the wrong version of golangci-lint installed, so can't debug locally - do you know what version I should have? I don't see it listed in public docs.)

**Comment by rileykarson** [comment]
`golangci-lint` has never worked super well for me locally, and the logs are a little harder to parse than we'd like for sure. This failed building, indicated by failing test as well as lint. Here's the error:

```
==> Checking source code against gofmt...
==> Checking that code complies with gofmt requirements...
gofmt needs running on the following files:
./google/resource_bigquery_table.go
./google/resource_bigquery_table_test.go
You can use the command: `make fmt` to reformat code.
GNUmakefile:22: recipe for target 'fmtcheck' failed
make: *** [fmtcheck] Error 1
==> installing required tooling...
```

**Comment by rileykarson** [comment]
That eventarc failure's not related

**Comment by gutschenritter** [comment]
Thanks Riley, I'm new to writing go code - ran `go fmt` on the two files I added and I *think* it fixed it. We'll see if the linter passes now.

**Comment by gutschenritter** [comment]
Believe the `TestAccEventarcTrigger_BasicHandWritten` test failure is unrelated.

**Comment by gutschenritter** [comment]
gentle ping

**Comment by gutschenritter** [comment]
@rileykarson gentle ping (sorry, know it was a long weekend)

File: mmv1/third_party/terraform/resources/resource_bigquery_table.go [hunk 1]
```diff
@@ -605,6 +605,16 @@ func resourceBigQueryTable() *schema.Resource {
...
 						},
+						// ConnectionId: [Optional] The connection specifying the credentials
+						// to be used to read external storage, such as Azure Blob,
+						// Cloud Storage, or S3. The connectionId can have the form
+						// "<project_id>.<location_id>.<connection_id>" or
+						// "projects/<project_id>/locations/<location_id>/connections/<connection_id>".
+						"connection_id": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: `The connection specifying the credentials to be used to read external storage, such as Azure Blob, Cloud Storage, or S3. The connectionId can have the form "<project_id>.<location_id>.<connection_id>" or "projects/<project_id>/locations/<location_id>/connections/<connection_id>".`,
```
Comment (rileykarson) [inline mmv1/third_party/terraform/resources/resource_bigquery_table.go:616]: You'll want to document the field under https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/website/docs/r/bigquery_table.html.markdown as well!

Comment (rileykarson) [inline mmv1/third_party/terraform/resources/resource_bigquery_table.go:616]: also nit: We've typically used `{{`/`}}` to show template strings in the provider, so I'd suggest that. As well, `project_id` and `location_id` generally have the `_id` dropped i.e.

`{{project}}.{{location}}.{{connection_id}}`

**Review by rileykarson — COMMENTED** [review]
One last thing, otherwise LGTM

File: mmv1/third_party/terraform/resources/resource_bigquery_table.go (hunk 1 above)
Comment (gutschenritter) [reply mmv1/third_party/terraform/resources/resource_bigquery_table.go]: Thanks Riley! Didn't know about this, done.

Comment (gutschenritter) [reply mmv1/third_party/terraform/resources/resource_bigquery_table.go]: done

**Pull request merged**